.github/workflows/registry-cleanup.yaml

```diff
@@ -0,0 +1,35 @@
+name: Remove old artifacts
+on:
+  schedule:
+    - cron: '0 12 * * *' # every day at 12:00 UTC
+  workflow_dispatch:
+
+jobs:
+  remove_old_artifacts:
+    name: Remove old artifacts
+    runs-on: ubuntu-latest
+    
+    permissions:
+      contents: read
+      packages: write
+    
+    timeout-minutes: 10 # stop the task if it takes longer
+
+    steps:
+      - name: Delete old package versions of nexus-configuration-controller
+        uses: actions/delete-package-versions@v5.0.0
+        with:
+          package-name: 'nexus-configuration-controller'
+          package-type: container
+          token: ${{ secrets.GITHUB_TOKEN }}
+          min-versions-to-keep: 10
+          delete-only-pre-release-versions: "true"
```

Comment:
This argument behaves weirdly, I would suggest to use a regex to limit versions affected by the pipeline: https://github.com/SneaksAndData/arcane-stream-sqlserver-change-tracking/blob/e6a34f331ced1854088ec3968dcf8effcefe2974/.github/workflows/cleanup-repository.yaml#L26